src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaTestHelpers.cs

```diff
@@ -35,14 +35,426 @@ public static void VerifyDisposed(SlhDsa slhDsa)
+        }
+
+        public static void AssertImportEncryptedPkcs8PrivateKey(
+            Action<ImportEncryptedPkcs8PrivateKeyCallback> test,
+            EncryptionPasswordType passwordTypeToTest = EncryptionPasswordType.All) =>
+            AssertImportEncryptedPkcs8PrivateKey(test, test, passwordTypeToTest);
+
+        internal delegate SlhDsa ImportEncryptedPkcs8PrivateKeyCallback(string password, ReadOnlySpan<byte> pkcs8);
+        public static void AssertImportEncryptedPkcs8PrivateKey(
+            Action<ImportEncryptedPkcs8PrivateKeyCallback> testDirectCall,
+            Action<ImportEncryptedPkcs8PrivateKeyCallback> testEmbeddedCall,
+            EncryptionPasswordType passwordTypeToTest = EncryptionPasswordType.All)
+        {
+            if ((passwordTypeToTest & EncryptionPasswordType.Char) != 0)
+            {
+                testDirectCall((password, pkcs8) => SlhDsa.ImportEncryptedPkcs8PrivateKey(password, pkcs8.ToArray()));
+                testDirectCall((password, pkcs8) => SlhDsa.ImportEncryptedPkcs8PrivateKey(password.AsSpan(), pkcs8));
+            }
+
+            if ((passwordTypeToTest & EncryptionPasswordType.Byte) != 0)
+            {
+                testDirectCall((password, pkcs8) =>
+                    SlhDsa.ImportEncryptedPkcs8PrivateKey(Encoding.UTF8.GetBytes(password), pkcs8.ToArray()));
+            }
+
+            AssertImportFromEncryptedPem(importPem =>
+            {
+                testEmbeddedCall((string password, ReadOnlySpan<byte> pkcs8) =>
+                {
+                    string pem = PemEncoding.WriteString("ENCRYPTED PRIVATE KEY", pkcs8);
+                    return importPem(pem, password);
+                });
+            });
+        }
+
+        internal delegate SlhDsa ImportFrpmEncryptedPemCallback(string source, string password);
+        public static void AssertImportFromEncryptedPem(
+            Action<ImportFrpmEncryptedPemCallback> callback,
+            EncryptionPasswordType passwordTypeToTest = EncryptionPasswordType.All)
+        {
+            if ((passwordTypeToTest & EncryptionPasswordType.Char) != 0)
+            {
+                callback(static (string pem, string password) => SlhDsa.ImportFromEncryptedPem(pem, password));
+                callback(static (string pem, string password) => SlhDsa.ImportFromEncryptedPem(pem.AsSpan(), password));
+            }
+
+            if ((passwordTypeToTest & EncryptionPasswordType.Byte) != 0)
+            {
+                callback(static (string pem, string password) =>
+                    SlhDsa.ImportFromEncryptedPem(pem, Encoding.UTF8.GetBytes(password)));
+                callback(static (string pem, string password) =>
+                    SlhDsa.ImportFromEncryptedPem(pem.AsSpan(), Encoding.UTF8.GetBytes(password)));
+            }
+        }
+
+        public static void AssertEncryptedPkcs8PrivateKeyContents(PbeParameters pbeParameters, ReadOnlyMemory<byte> contents)
```

Comment:
The fact that this doesn't have any SlhDsa typed parameters suggests this belongs in a higher level helpers type, so it could be reused by other algorithms.